src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -865,13 +868,78 @@ internal static string ResolvePackagePath(string searchDirectory, string searchP
             {
                 // If the search does not contain wild cards, and the target path shares the same extension, copy it
                 // e.g. <file src="ie\css\style.css" target="Content\css\ie.css" /> --> Content\css\ie.css
-                return targetPath;
+                packagePath = targetPath;
+                targetPath = null;
             }
             else
             {
                 packagePath = Path.GetFileName(fullPath);
             }
-            return Path.Combine(targetPath ?? String.Empty, packagePath);
+
+            string path = Path.Combine(targetPath ?? string.Empty, packagePath);
+
+            // Translate the tfm alias to its actual framework if necessary.
+            foreach (string knownFolder in PackagingConstants.Folders.Known)
+            {
+                string folderPrefix = knownFolder + Path.DirectorySeparatorChar;
+                if (path.Length > folderPrefix.Length &&
+                    path.StartsWith(folderPrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    string frameworkPart = null;
+                    if (knownFolder == PackagingConstants.Folders.ContentFiles && path.Count(sep => sep == Path.DirectorySeparatorChar) > 2)
+                    {
+                        // ContentFiles might have an extra level of nesting for the
+                        // language, so we have to skip over _that_ in order
+                        // to get to the framework itself.
+                        var languageEndIndex = path.IndexOf(System.IO.Path.DirectorySeparatorChar, folderPrefix.Length + 1);
+                        frameworkPart = path.Substring(folderPrefix.Length + (languageEndIndex - folderPrefix.Length));
+                    }
+                    else
+                    {
+                        frameworkPart = path.Substring(folderPrefix.Length);
+                    }
+
+                    string aliasString = Path.GetDirectoryName(frameworkPart).Split(Path.DirectorySeparatorChar).First();
+
+                    if (string.IsNullOrEmpty(aliasString))
+                    {
+                        break;
+                    }
+
+                    string frameworkString = null;
+
+                    NuGetFramework framework = null;
+
+#pragma warning disable CS0618 // This API is only ever meant to be used temporarily inside NuGet itself, not outside of our SDK
+                    if (AliasFolderNameMapping != null && AliasFolderNameMapping.TryGetValue(aliasString, out frameworkString))
+#pragma warning restore CS0618
+                    {
+                        path = folderPrefix + frameworkString + Path.DirectorySeparatorChar + path.Substring(folderPrefix.Length + aliasString.Length + 1);
```

Comment:
I'm worried this doesn't correctly use the right folderPrefix for `contentFiles` or `runtimes`.